cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -152,6 +152,25 @@ size_t batched_decompress_temp_size(compression_type compression,
   return temp_size;
 }
 
+void check_is_zstd_enabled()
+{
+  CUDF_EXPECTS(NVCOMP_HAS_ZSTD_DECOMP, "nvCOMP 2.3 or newer is required for Zstandard compression");
+  CUDF_EXPECTS(NVCOMP_ZSTD_IS_STABLE or cudf::io::detail::nvcomp_integration::is_all_enabled(),
+               "Zstandard compression is experimental, you can enable it through "
+               "`LIBCUDF_NVCOMP_POLICY` environment variable.");
+
+#if NVCOMP_ZSTD_IS_DISABLED_ON_PASCAL
+  int device;
+  int cc_major;
+  int cc_minor;
+  CUDF_CUDA_TRY(cudaGetDevice(&device));
+  CUDF_CUDA_TRY(cudaDeviceGetAttribute(&cc_major, cudaDevAttrComputeCapabilityMajor, device));
+  CUDF_CUDA_TRY(cudaDeviceGetAttribute(&cc_minor, cudaDevAttrComputeCapabilityMinor, device));
+  CUDF_EXPECTS(cc_major != 6 or cc_minor != 1,
+               "Zstandard decompression is disabled on some Pascal GPUs");
```

Comment:
Is the issue only on `6.1`? will `6.0` work fine?

Reply:
Yup. No repro on P100, which is compute capability 6.0.

Reply:
I see, got it. Thanks 👍